### PR TITLE
Try to coerce deps names for non-identifier keys

### PR DESCRIPTION
**Pull request by lundibundi**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

File: lib/dependencies.js [hunk 1]
```diff
@@ -21,6 +21,25 @@ const dependencies = [...internals, ...npmpkg, ...metapkg];
...
+    name = name
+      .split(/[/-]/)
+      .map((sub, i) => {
+        if (i === 0) return sub;
+        return sub.slice(0, 1).toUpperCase() + sub.slice(1);
+      })
+      .join('');
+    container[name] = value;
+  }
+};
```
Comment (tshemsedinov) [inline lib/dependencies.js:41]:
```suggestion
const spinalToCamel = (name) => {
  const words = name.split('-');
  const first = words.shift();
  return first + words.map(toUpperCamel).join('');
};

const assignNamespace = (container, name, value) => {
  container[name] = value;
  let key = name.startsWith('@') ? name.slice(1) : name;
  key = replace(key, '/', '-');
  key = spinalToCamel(key);
  container[key] = value;
};
```
p.s. We have `replace` and `toUpperCamel` in `metautils`

Comment (tshemsedinov) [reply lib/dependencies.js:41]: And it would be good to put `spinalToCamel` to `metautils` as well

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

File: lib/dependencies.js (hunk 1 above)
Comment (tshemsedinov) [reply lib/dependencies.js:41]: `spinalToCamel` is now improved, landed but not published to npm

Comment (lundibundi) [reply lib/dependencies.js]: Pushed suggested changes with a few improvements for now.

**Comment by tshemsedinov** [comment]
@lundibundi now we have `spinalToCamel` in `metautil 3.5.17`

**Comment by lundibundi** [comment]
@tshemsedinov updated.

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/impress/commit/5bed24a956f75c046a696890a1e11659c960a1bc

**Pull request closed without merging**